### PR TITLE
Add Account cache

### PR DESCRIPTION
**Pull request by laurb9**
Summary
=======

Optional cache for Accounts. This cache is not necessary for the current static file fetcher, it is intended to support upcoming http API and db fetchers.

Implementation
===========
Added [cache support for account fetcher](https://github.com/prebid/prebid-server/compare/master...openx:account-cache?expand=1#diff-c705d31b368d04ad30e87727936db388R194-R204). 
Add [new `size_bytes` parameter for `in_memory_cache`](https://github.com/prebid/prebid-server/compare/master...openx:account-cache?expand=1#diff-64d858641d4b8eb33576093f89ca1c21R400-R401) to control single cache.
Add necessary config validation so `accounts` uses `size_bytes` while the rest have `request_cache_size_bytes`
 and `imp_cache_size_bytes`.
Removed some redundant tests and refactored others a little.

Example config:
```yaml
accounts:
  in_memory_cache:
    type: lru
    ttl_seconds: 10
    size_bytes: 500000
```

### CODE REVIEW

File: stored_requests/fetcher.go [hunk 1]
```diff
@@ -190,8 +191,17 @@ func (f *fetcherWithCache) FetchRequests(ctx context.Context, requestIDs []strin
 	return
 }
 
-func (f *fetcherWithCache) FetchAccount(ctx context.Context, accountID string) (json.RawMessage, []error) {
-	return f.fetcher.FetchAccount(ctx, accountID)
+func (f *fetcherWithCache) FetchAccount(ctx context.Context, accountID string) (account json.RawMessage, errs []error) {
+	accountData := f.cache.Accounts.Get(ctx, []string{accountID})
+	// TODO: add metrics
```
Comment (bsardo) [inline stored_requests/fetcher.go:196]: You mentioned the metrics will be a separate PR right? And you're also planning to add account HTTP and DB fetchers in their own PRs too?

**Review by bsardo — COMMENTED** [review]
This is looking good.

File: stored_requests/fetcher.go (hunk 1 above)
Comment (laurb9) [reply stored_requests/fetcher.go:196]: The metrics commit is ready, I can make a PR right after this, or add another commit here.
Account http is almost ready, will make a PR separately.
I will not have time this round for the Db fetcher, but I'll try to take a stab at it.

**Review by bsardo — DISMISSED** [review]

File: config/stored_requests.go [hunk 2]
```diff
@@ -384,32 +397,57 @@ type InMemoryCache struct {
...
-			errs = append(errs, fmt.Errorf("%s: in_memory_cache.request_cache_size_bytes must be >= 0 when in_memory_cache.type=lru. Got %d", section, cfg.RequestCacheSize))
-		}
-		if cfg.ImpCacheSize <= 0 {
-			errs = append(errs, fmt.Errorf("%s: in_memory_cache.imp_cache_size_bytes must be >= 0 when in_memory_cache.type=lru. Got %d", section, cfg.ImpCacheSize))
+		if dataType == AccountDataType {
+			// single cache
+			if cfg.Size <= 0 {
+				errs = append(errs, fmt.Errorf("%s: in_memory_cache.size_bytes must be >= 0 when in_memory_cache.type=lru. Got %d", section, cfg.Size))
+			}
+			if cfg.RequestCacheSize > 0 || cfg.ImpCacheSize > 0 { // should this be a warning instead ? "field is ignored"
```
Comment (mansinahar) [inline config/stored_requests.go:437]: Yeah, I think this should be fine with being a warning but either way, we should keep this consistent with what we do when `size` is specified in case of stored request/imp cache which right now is a warning

**Review by mansinahar — COMMENTED** [review]
LGTM other than the minor comment above

File: config/stored_requests.go (hunk 2 above)
Comment (laurb9) [reply config/stored_requests.go]: I should have changed both at the same time, sorry.

Comment (mansinahar) [reply config/stored_requests.go]: No worries! Happens to all of us. Thank you for the PR :)

**Review by bsardo — APPROVED** [review]
